### PR TITLE
Change footer behavior on mobile #1258

### PR DESCRIPTION
**Pull request by ithiame**

### CODE REVIEW

**Comment by ithiame** [comment]
Before:
image

After :
image

**Comment by coyotte508** [comment]
I think we only want to change the behavior of the links -"terms of service", ... in your screenshot cc @Tirodem 

no need for such a large commit, you can change css classnames locally

**Comment by Tirodem** [comment]
> I think we only want to change the behavior of the links -"terms of service", ... in your screenshot cc @Tirodem
> 
> no need for such a large commit, you can change css classnames locally

50/50, I forgot about "OUR SHOP".

So instead of having
[OUR SITE][OUR SHOP]
[OUR CONTACT]

It's indeed interesting to have
[OUR SITE]
[OUR SHOP]
[OUR CONTACT] on mobile

BTW, @ithiame , links are not displayed as I expected.
image
Your version display footers links on the same line
[link1] [link2] [link3] [link4] [link5]
I expected them to be like that :
[link1]
[link2]
[link3]
[link4]
[link5]

**Comment by coyotte508** [comment]
ok, still, no need to copy paste everything, a few changes here and there are all that needed

For CMS pages it was different because the pages can have completely different content on mobile & desktop

File: src/routes/(app)/+layout.svelte [hunk 1]
```diff
@@ -555,7 +564,14 @@
...
 				{:else if data.displayPoweredBy}
-					<div class="flex w-full">
+					<div class="w-full hidden sm:flex">
+						<a class="items-center gap-4" href="https://be-bop.io" target="_blank">
+							<span class="font-light">{t('footer.poweredBy')} </span>
+							<img class="h-[40px] w-auto hidden dark:inline" src={DEFAULT_LOGO} alt="" />
+							<img class="h-[40px] w-auto dark:hidden" src={DEFAULT_LOGO_DARK} alt="" />
+						</a>
+					</div>
+					<div class="sm:hidden justify-center flex w-full">
```
Comment (coyotte508) [inline src/routes/(app)/+layout.svelte:574]:
```suggestion
					<div class="justify-center sm:justify-normal flex w-full">
```

no?

and then no need for copy paste

File: src/routes/(app)/+layout.svelte [hunk 2]
```diff
@@ -531,8 +531,17 @@
...
+					<div class="hidden sm:flex flex-row gap-2">
+						{#each data.links.footer as link}
+							<a
+								href={link.href}
+								target={link.href.startsWith('http') ? '_blank' : '_self'}
+								data-sveltekit-preload-data="off">{link.label}</a
+							>
+						{/each}
+					</div>
+					<div class="sm:hidden flex items-end flex-col gap-2">
```
Comment (coyotte508) [inline src/routes/(app)/+layout.svelte:544]:
```suggestion
					<div class="flex items-end flex-col sm:flex-row gap-2">
```

no?

File: src/routes/(app)/+layout.svelte [hunk 3]
```diff
@@ -476,7 +476,7 @@
 
 		<footer class="footer h-auto items-center flex print:hidden">
 			<div
-				class="mx-auto max-w-7xl px-6 py-6 flex items-start justify-between gap-2 gap-y-8 w-full flex-wrap"
+				class="mx-auto max-w-7xl px-6 py-6 grid items-start justify-between gap-y-8 w-full grid-cols-1 lg:grid-cols-4 gap-4"
```
Comment (coyotte508) [inline src/routes/(app)/+layout.svelte:479]: `lg` is `1024px`

Anyway let's keep the behavior the same for large screens? You can do things like `grid grid-cols-1 lg:flex lg:flex-wrap`

File: src/routes/(app)/+layout.svelte [hunk 4]
```diff
@@ -476,7 +476,7 @@
 
 		<footer class="footer h-auto items-center flex print:hidden">
 			<div
-				class="mx-auto max-w-7xl px-6 py-6 flex items-start justify-between gap-2 gap-y-8 w-full flex-wrap"
+				class="mx-auto max-w-7xl px-6 py-6 items-start justify-between gap-y-8 w-full grid grid-cols-1 lg:flex lg:flex-wrap gap-4"
```
Comment (coyotte508) [inline src/routes/(app)/+layout.svelte:479]:
```suggestion
				class="mx-auto max-w-7xl px-6 py-6 items-start justify-between gap-y-8 w-full grid grid-cols-1 lg:flex lg:flex-wrap gap-4 lg:gap-2"
```

**Comment by coyotte508** [comment]
Can you make sure everything's clean - previous behavior 100 % identical for larger screens?

Also, it's wierd having `lg:` (1000px screens) and `sm:` (642px screens) in the same PR, is it intended?

**Comment by ithiame** [comment]
As i understand we handle screen >= 1024 and screen < 1024

so i think we can juste use lg

**Review by coyotte508 — COMMENTED** [review]
~ok for me, feel free to merge if @Tirodem is ok

**Review by Tirodem — APPROVED** [review]

**Pull request merged**